grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java

```diff
@@ -233,17 +233,16 @@ private static void attachSingleGeneratedRelation(Representation currentHal, Con
 
     static void generateConceptState(Representation resource, Concept concept){
 
+        resource.withProperty(ID_PROPERTY, concept.getId().getValue());
+
         if (concept.isInstance()) {
             Instance instance = concept.asInstance();
-            resource.withProperty(ID_PROPERTY, instance.getId().getValue())
-                    .withProperty(TYPE_PROPERTY, instance.type().getName().getValue())
+            resource.withProperty(TYPE_PROPERTY, instance.type().getName().getValue())
                     .withProperty(BASETYPE_PROPERTY, getBaseType(instance).name());
         } else {
-            Type type = concept.asType();
-            resource.withProperty(ID_PROPERTY, type.getId().getValue())
-                    .withProperty(BASETYPE_PROPERTY, getBaseType(type).name());
-
+            resource.withProperty(BASETYPE_PROPERTY, getBaseType(concept.asType()).name());
```

Comment:
it's the same method but it's overloaded, so I guess we need the else block? we are performing different checks in the two `getBaseType` methods